src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProject.cs

```diff
@@ -879,78 +884,107 @@ public void AddAnalyzerReference(string fullPath)
+        public void RemoveAnalyzerReference(string fullPathX)
         {
-            if (string.IsNullOrEmpty(fullPath))
+            if (string.IsNullOrEmpty(fullPathX))
             {
-                throw new ArgumentException("message", nameof(fullPath));
+                throw new ArgumentException("message", nameof(fullPathX));
             }
 
             using (_gate.DisposableWait())
             {
-                if (!_analyzerPathsToAnalyzers.TryGetValue(fullPath, out var visualStudioAnalyzer))
+                foreach (var mappedFullPath in GetMapedAnalyzerPaths(fullPathX))
                 {
-                    throw new ArgumentException($"'{fullPath}' is not an analyzer of this project.", nameof(fullPath));
-                }
+                    if (!_analyzerPathsToAnalyzers.TryGetValue(mappedFullPath, out var visualStudioAnalyzer))
+                    {
+                        throw new ArgumentException($"'{mappedFullPath}' is not an analyzer of this project.", nameof(mappedFullPath));
+                    }
 
-                _analyzerPathsToAnalyzers.Remove(fullPath);
+                    _analyzerPathsToAnalyzers.Remove(mappedFullPath);
 
-                if (_activeBatchScopes > 0)
-                {
-                    // This analyzer may be one we've just added in the same batch; in that case, just don't add
-                    // it in the first place.
-                    if (_analyzersAddedInBatch.Remove(visualStudioAnalyzer))
+                    if (_activeBatchScopes > 0)
                     {
-                        // Nothing is holding onto this analyzer now, so get rid of it
-                        visualStudioAnalyzer.Dispose();
+                        // This analyzer may be one we've just added in the same batch; in that case, just don't add
+                        // it in the first place.
+                        if (_analyzersAddedInBatch.Remove(visualStudioAnalyzer))
+                        {
+                            // Nothing is holding onto this analyzer now, so get rid of it
+                            visualStudioAnalyzer.Dispose();
+                        }
+                        else
+                        {
+                            _analyzersRemovedInBatch.Add(visualStudioAnalyzer);
+                        }
                     }
                     else
                     {
-                        _analyzersRemovedInBatch.Add(visualStudioAnalyzer);
+                        _workspace.ApplyChangeToWorkspace(w => w.OnAnalyzerReferenceRemoved(Id, visualStudioAnalyzer.GetReference()));
+
+                        visualStudioAnalyzer.Dispose();
                     }
                 }
-                else
+            }
+        }
+
+        private IEnumerable<string> GetMapedAnalyzerPaths(string fullPath)
+        {
+            if (fullPath.Split(s_directorySeparator) is [.., "Sdks", "Microsoft.NET.Sdk.Razor", "source-generators", var fileName])
```

Comment:
The dependency on path casing worries me a bit. Essentially if one team changes the casing of anything here then suddenly our solution breaks down. It shouldn't happen ideally but it has happened a few times by accident. We made such a mistake fixing up razor this summer. 

Is there a reason we can't just depend on the file name only vs. the path?

Reply:
We need to handle dependencies of the generator dll. Replace the ones referenced by the SDK version with those referenced by the VSIX version.

Reply:
We can make the path comparison case-insensitive.